spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java

```diff
@@ -63,4 +63,14 @@ private SparkSQLProperties() {}
   // Controls the WAP branch used for write-audit-publish workflow.
   // When set, new snapshots will be committed to this branch.
   public static final String WAP_BRANCH = "spark.wap.branch";
+
+  // This property doesn't need to be transferred to Spark 3.4 because we have already set
```

Comment:
This description seems to just describe the implementation, what we would need here is just what an end user would need to know. What does this property do for the user and why should they change it?

I would probably change the whole name to something like, "ENABLE_MERGE_AQE" or something like that.
"spark.merge-aqe.enabled" defaulting to false.

Reply:
@aokolnychyi what do you think?

Reply:
I believe this would apply to all writes, not only row-level operations, if extensions are enabled?
What about `spark.sql.iceberg.write-aqe.enabled` with `false` by default? I am not sure about `write-aqe` part of it but I think it has be generic and start with `spark.sql.iceberg` prefix.

Also, the comment is probably too specific, I agree with @RussellSpitzer it should be for the user.

Reply:
I agree that this applies to all writes, not just row level operations. `spark.sql.iceberg.write-aqe.enabled` sounds like a good option.